homeassistant/components/heos/__init__.py

```diff
@@ -331,9 +333,13 @@ def connect_update(self):
             heos_const.SIGNAL_HEOS_EVENT, self.async_update_groups
         )
 
-    def force_update_groups(self):
-        """Update group membership, regardless of any HEOS event."""
-        self._hass.add_job(self.async_update_groups, heos_const.EVENT_GROUPS_CHANGED)
+        # When adding a new HEOS player we need to update the groups.
+        async def _async_handle_player_added():
+            await self.async_update_groups(SIGNAL_HEOS_PLAYER_ADDED)
+
+        self._hass.helpers.dispatcher.async_dispatcher_connect(
```

Comment:
Please import the `async_dispatcher_connect` instead of using it on `hass.helpers`. The latter is legacy.

Reply:
Thanks for pointing that out. I am importing `async_dispatcher_connect`/`async_dispatcher_send` now.